batch/plugins/org.jboss.tools.batch.ui/src/org/jboss/tools/batch/ui/quickfixes/BatchQuickFixGenerator.java

```diff
@@ -0,0 +1,262 @@
+					new CreateBatchArtifactQuickFix(file, text, BatchArtifactType.STEP_LISTENER, BatchQuickFixMessages.CREATE_STEP_LISTENER)
+				};
+			}
+		}else if(messageId == BatchValidator.DECIDER_IS_NOT_FOUND_ID){
+			IJavaElement element = findJavaElementByQualifiedName(javaProject, text);
+			if(element == null){
+				return new IQuickFix[] {
+					new CreateBatchArtifactQuickFix(file, text, BatchArtifactType.DECIDER, BatchQuickFixMessages.CREATE_DECIDER)
+				};
+			}
+		}else if(messageId == BatchValidator.CHECKPOINT_ALGORITHM_IS_NOT_FOUND_ID){
+			IJavaElement element = findJavaElementByQualifiedName(javaProject, text);
+			if(element == null){
+				return new IQuickFix[] {
+					new CreateBatchArtifactQuickFix(file, text, BatchArtifactType.CHECKPOINT_ALGORITHM, BatchQuickFixMessages.CREATE_CHECKPOINT_ALGORYTHM)
+				};
+			}
+		}else if(messageId == BatchValidator.ITEM_READER_IS_NOT_FOUND_ID){
+			IJavaElement element = findJavaElementByQualifiedName(javaProject, text);
+			if(element == null){
+				return new IQuickFix[] {
+					new CreateBatchArtifactQuickFix(file, text, BatchArtifactType.ITEM_READER, BatchQuickFixMessages.CREATE_ITEM_READER)
+				};
+			}
+		}else if(messageId == BatchValidator.ITEM_WRITER_IS_NOT_FOUND_ID){
+			IJavaElement element = findJavaElementByQualifiedName(javaProject, text);
+			if(element == null){
+				return new IQuickFix[] {
+					new CreateBatchArtifactQuickFix(file, text, BatchArtifactType.ITEM_WRITER, BatchQuickFixMessages.CREATE_ITEM_WRITER)
+				};
+			}
+		}else if(messageId == BatchValidator.ITEM_PROCESSOR_IS_NOT_FOUND_ID){
+			IJavaElement element = findJavaElementByQualifiedName(javaProject, text);
+			if(element == null){
+				return new IQuickFix[] {
+					new CreateBatchArtifactQuickFix(file, text, BatchArtifactType.ITEM_PROCESSOR, BatchQuickFixMessages.CREATE_ITEM_PROCESSOR)
+				};
+			}
+		}else if(messageId == BatchValidator.MAPPER_IS_NOT_FOUND_ID){
+			IJavaElement element = findJavaElementByQualifiedName(javaProject, text);
+			if(element == null){
+				return new IQuickFix[] {
+					new CreateBatchArtifactQuickFix(file, text, BatchArtifactType.PARTITION_MAPPER, BatchQuickFixMessages.CREATE_PARTITION_MAPPER)
+				};
+			}
+		}else if(messageId == BatchValidator.ANALYZER_IS_NOT_FOUND_ID){
+			IJavaElement element = findJavaElementByQualifiedName(javaProject, text);
+			if(element == null){
+				return new IQuickFix[] {
+					new CreateBatchArtifactQuickFix(file, text, BatchArtifactType.PARTITION_ANALYZER, BatchQuickFixMessages.CREATE_PARTITION_ANALYZER)
+				};
+			}
+		}else if(messageId == BatchValidator.COLLECTOR_IS_NOT_FOUND_ID){
+			IJavaElement element = findJavaElementByQualifiedName(javaProject, text);
+			if(element == null){
+				return new IQuickFix[] {
+					new CreateBatchArtifactQuickFix(file, text, BatchArtifactType.PARTITION_COLLECTOR, BatchQuickFixMessages.CREATE_PARTITION_COLLECTOR)
+				};
+			}
+		}else if(messageId == BatchValidator.REDUCER_IS_NOT_FOUND_ID){
```

Comment:
IMHO this code would look clearer if you used something like this:
```java
switch (messageId) {
            case BatchValidator.BATCHLET_IS_NOT_FOUND_ID:  return getQuickFixes(javaProject, text, BatchArtifactType.BATCHLET, BatchQuickFixMessages.CREATE_BATCHLET);
            case BatchValidator.JOB_LISTENER_IS_NOT_FOUND_ID:  return getQuickFixes(javaProject, text, BatchArtifactType.JOB_LISTENER, BatchQuickFixMessages.CREATE_JOB_LISTENER);
...
            default: return new IQuickFix[0];
                     break;
}



IQuickFix[] getQuickFixes(IJavaProject javaProject, String text, BatchArtifactType type, String message) {
	IJavaElement element = findJavaElementByQualifiedName(javaProject, text);
	if(element == null){
		return new IQuickFix[] {
			new CreateBatchArtifactQuickFix(file, text, type, message)
		};
	}
	return new IQuickFix[0];
}
```

Or the same but with if(){...} else if(){...} if you like it better than switch.